src/ragger/backend/interface.py

```diff
@@ -287,6 +293,24 @@ def both_click(self) -> None:
         """
         raise NotImplementedError
 
+    def navigate(self, instruction: NavigationInstruction) -> None:
+        """
+        Performs a click on buttons of the device depending of the navigation instruction.
+
+        :raises NotImplementedError: If the navigation instruction is not implemented.
+
+        :return: None
+        :rtype: NoneType
+        """
+        if instruction == NavigationInstruction.GO_TO_NEXT_SCREEN:
```

Comment:
@lpascal-ledger do you think this should be implemented in this way (in the interface directly) ?

Reply:
@xchapron-ledger actually I think the navigation method should be given a list of instructions and loop through them. Additionnaly, like navigate until snap , a "takesnaps" boolean parameter should be given to decide whether or not to take snapshots or not at every step.

Then , navigate and compare would just pass its own navigation instructions to navigate.

Reply:
Ok